test/unit/components/shepherd-button.spec.js

```diff
@@ -93,6 +93,33 @@ describe('component/ShepherdButton', () => {
       expect(button).toHaveAttribute('aria-label', '5');
     });
 
+    it('label - funtion', () => {
+      const labelFunction = (label) => label;
+      const config = {
+        label: labelFunction('Test')
+      };
+
+      const { container, rerender } = render(ShepherdButton, {
+        props: {
+          config
+        }
+      });
+
+      const button = container.querySelector('.shepherd-button');
+      expect(button).toHaveAttribute('aria-label', 'Test');
+
+      config.label = labelFunction('Test 2');
+
+      rerender({
+        props: {
+          config
+        }
+      });
+
+      const buttonUpdated = container.querySelector('.shepherd-button');
+      expect(buttonUpdated).toHaveAttribute('aria-label', 'Test 2');
```

Comment:
```suggestion
      let label = 'Test'
      const labelFunction = () => label;
      const config = {
        label: labelFunction
      };

      const { container, rerender } = render(ShepherdButton, {
        props: {
          config
        }
      });

      const button = container.querySelector('.shepherd-button');
      expect(button).toHaveAttribute('aria-label', 'Test');

      label = 'Test 2';

      rerender({
        props: {
          config
        }
      });

      const buttonUpdated = container.querySelector('.shepherd-button');
      expect(buttonUpdated).toHaveAttribute('aria-label', 'Test 2');
```

@radibit this is what I meant here, changing the value the function returns. The test was calling the functions manually, so you were actually just setting strings here.

Reply:
@rwwagner90 You're absolutely right 🙈 My mistake since the tests were indeed obsolete in that form. I've updated both test suites with the last commit [59c323e](https://github.com/shipshapecode/shepherd/pull/1815/commits/59c323ecc7ec68393debfb193c3df2b39ffe5d65)